### PR TITLE
MM-32962/MM-32930 Dependency updates

### PR DESCRIPTION
**Pull request by hmhealey**
Most of this month's dependency updates. There's a few bigger ones we're still avoiding and a couple I'll be creating tickets for.

The actual code changes are due to ESLint and TypeScript updates

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-32962
https://mattermost.atlassian.net/browse/MM-32930

### CODE REVIEW

File: components/threading/channel_threads/thread_footer/__snapshots__/thread_footer.test.tsx.snap [hunk 1]
```diff
@@ -630,7 +630,7 @@ exports[`components/threading/channel_threads/thread_footer should report total
       <injectIntl(Timestamp)
         day="numeric"
         dispatch={[Function]}
-        hour12={true}
+        hourCycle="h12"
```
Comment (calebroseland) [inline components/threading/channel_threads/thread_footer/__snapshots__/thread_footer.test.tsx.snap:633]: Has anything change recently with Intl/ICU support in Jest envs? I was trying to trace down why `hourCycle` support was missing (why this change in thread_footer.test.tsx.snap keeps flipping)

I think our current Jest env/setup + Node 10 causes missing `DateTimeFormat...hourCycle` support. I'm guessing it runs fine on Node 14.

**Review by crspeller — APPROVED** [review]

File: components/threading/channel_threads/thread_footer/__snapshots__/thread_footer.test.tsx.snap (hunk 1 above)
Comment (hmhealey) [reply components/threading/channel_threads/thread_footer/__snapshots__/thread_footer.test.tsx.snap:633]: It's Node 14 vs Node 10. I've been running Node 14 locally, so that's why the update changed this. That's partly what's breaking CI right now though.

File: package.json [hunk 2]
```diff
@@ -16,174 +16,174 @@
...
     "eslint-plugin-mattermost": "github:mattermost/eslint-plugin-mattermost#46ad99355644a719bf32082f472048f526605181",
-    "eslint-plugin-no-only-tests": "2.4.0",
-    "eslint-plugin-react": "7.22.0",
+    "eslint-plugin-no-only-tests": "2.5.0",
+    "eslint-plugin-react": "7.23.2",
     "file-loader": "6.2.0",
-    "full-icu": "1.3.1",
-    "html-loader": "1.3.2",
-    "html-webpack-plugin": "5.0.0",
+    "full-icu": "1.3.4",
```
[the remark this answers is not in the document]
Comment (hmhealey) [reply package.json:157]: Yeah, I was planning on it once I got around to upgrading CI to Node 14. Looks like I need to do that sooner rather than later though

**Comment by hmhealey** [comment]
Moving back to WIP since I'm going to need to address the problems of CI being on Node 10 and storybook being broken now.

**Comment by hmhealey** [comment]
Added `@storybook/builder-webpack5`, `dotenv-webpack`, and `terser-webpack-plugin` as explicit dependencies so that Storybook will support Webpack 5. The first one is the only one that should be needed for that, but the it was previously using old versions of the other two which was causing errors.

**Comment by hmhealey** [comment]
I also reverted the snapshots which were modified by Node 14. Those can be changed once we get CI running Node 14 as well.

**Review by calebroseland — APPROVED** [review]

**Review by jgilliam17 — APPROVED** [review]
Thanks @hmhealey 
Tested, looks good to merge.
- Short smoke test and verified [E2E test results](https://mm-cypress-report.s3.amazonaws.com/4576-webapp-pr-7929-mattermost-enterprise-edition:383ce8b/mochawesome.html), manually re-tested 12 failed tests that differ from master test run - all pass.
Also thanks in advance for adding a follow-up issue for plugin error found during testing. MM-35056
